### PR TITLE
Establish UI Component Test Coverage and Testing Strategy Foundation (Vitest, Testing Library)

### PR DESCRIPTION
**Pull request by lapidix**
### Overview
This project applies a comprehensive UI-centric testing strategy, focusing on React component behavior, user interaction, and accessibility. The current coverage includes major UI components and pages, ensuring visual and functional correctness.

#### Test Framework
- Stack: Vitest + Testing Library (React, Jest-DOM)
- Dependencies: vitest, @testing-library/react, @testing-library/jest-dom, etc.
- Global setup via `setup.ts` (includes React global import and console mocking)

#### UI Test Scope
- Targets all key UI components and pages:
  - Entities: User, Post, Comment (component level)
  - Widgets: Header, PostList, PostDetail, CommentForm, etc.
  - Pages: Home, PostDetail, etc.
- Focus:
  - Rendering scenarios (with/without data, edge cases)
  - User interactions (click, input, accessibility)
  - Custom props, styling, and conditional rendering

#### Test Utilities & Structure
- Centralized test fixtures and helpers for mock data and reusable logic
- Standardized directory structure (`__tests__` folders per feature/component)
- Consistent test naming and organization for maintainability

#### Configuration
- Test scripts for run, watch, and coverage in `package.json`
- Test-related settings in `.gitignore` and config files
- Coverage reporting via Vitest

### Motivation
- Ensure UI correctness and prevent regression during refactoring or feature development
- Validate accessibility and user experience in core flows

### CODE REVIEW

File: src/features/comment/ui/comment-view/CommentView.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { CommentDto } from "@/entities/comment/infrastructure/dto";
-import { UserAvatar } from "@/entities/user/ui/identifier";
-import { formatDate } from "@/shared/libs/date";
+import { CommentDto } from "@/entities/comment";
+import { UserAvatar } from "@/entities/user";
```
[the remark this answers is not in the document]
Comment (lapidix) [reply src/features/comment/ui/comment-view/CommentView.tsx:2]: @gemini-code-assist The barrel file at `src/entities/user/index.ts` does export `UserAvatar`. You can import `UserAvatar` directly from the user entity module, as it is explicitly re-exported there. This makes it accessible for use in other parts of the application without needing to reference the deeper file path.

File: src/features/comment/__tests__/ui/comment-form/CommentForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+      // When: Render form and submit it
+      const { container } = render(
+        <FullWrapper>
+          <CommentForm userProfileImage={userProfileImage} />
+        </FullWrapper>
+      );
+
+      const formElement = container.querySelector("form");
+      fireEvent.submit(formElement!);
```
[the remark this answers is not in the document]
Comment (lapidix) [reply src/features/comment/__tests__/ui/comment-form/CommentForm.test.tsx]: @gemini-code-assist The code in question is meant to serve as an example, so the form submission logic was intentionally left empty. The disabled state and lack of actual submission handling are by design for demonstration purposes only.

File: src/features/comment/ui/comment-view/CommentView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lapidix) [reply src/features/comment/ui/comment-view/CommentView.tsx:2]: @gemini-code-assist Please double-check the current state of `src/entities/user/index.ts`.  includes the following exports:

```ts
export { User, UserFactory, type UserRepository } from "./core";
export { USER_QUERY_KEYS } from "./infrastructure/api";
export { type UserDto, type UserProfileDto } from "./infrastructure/dto";
export { UserApiRepository } from "./infrastructure/repository";
export { UserMapper } from "./mapper";
export { type UserEntity } from "./types";
export { UserAvatar, UserIdentifier } from "./ui/identifier";
```

As shown above, both `UserAvatar` and `UserIdentifier` are now explicitly exported from the barrel file. Please confirm based on the latest code.

[the remark this answers is not in the document]
Comment (lapidix) [reply src/features/comment/ui/comment-view/CommentView.tsx:2]: @gemini-code-assist The exports for `UserAvatar` and `UserIdentifier` were not newly added or recently modified—they have been present in `src/entities/user/index.ts` from the beginning. Please double-check the base branch or the initial state of the pull request to confirm; these exports were not introduced as part of a recent change.

**Comment by lapidix** [comment]
@gemini-code-assist review
